multicore_runtime/static_files.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Static file serving functionality invoked by wsgi_config.py."""
+
+import logging
+import mimetypes
+import os
+import re
+
+from werkzeug.wrappers import Request
+from werkzeug.wrappers import Response
+from werkzeug.wsgi import wrap_file
+
+
+def static_app_for_regex_and_files(regex, files, upload, mime_type=None):
+  """Returns a WSGI app that serves static files.
+
+  Args:
+    regex: A url-matching regex as specified in appinfo.URLMap.
+    files: A static_files definition as specified in appinfo.URLMap.
+      May include a regex backref. See the appinfo.URLMap docstring
+      for more information.
+    upload: A filename-matching regex as specified in appinfo.URLMap.
+      Only files that match this regex will be served regardless of other
+      arguments.
+    mime_type: A mime type to apply to all files. If absent,
+      mimetypes.guess_type() is used.
+
+  Returns:
+    A static file-serving WSGI app closed over the inputs.
+  """
+  @Request.application  # Transforms wsgi_env, start_response args into request
+  def serve_static_files(request):
+    """Serve a static file."""
+    # First, match the path against the regex...
+    matcher = re.match(regex, request.path)
```

Comment:
In order to get this far, the request had to match the regex in the dispatcher, so the match here is just to get something we can apply a backref to.

However, now that you mention it, I don't like that this function is vulnerable if it is taken out of its intended context.